src/mon/PGMonitor.cc

```diff
@@ -1375,8 +1375,11 @@ void PGMonitor::dump_object_stat_sum(TextTable &tbl, Formatter *f,
     tbl << stringify(si_t(sum.num_bytes));
     int64_t kb_used = SHIFT_ROUND_UP(sum.num_bytes, 10);
     float used = 0.0;
-    if (pg_map.osd_sum.kb > 0)
-      used = (float)kb_used * raw_used_rate * curr_object_copies_rate / pg_map.osd_sum.kb;
+    if (avail) {
+      used = static_cast<float>(sum.num_bytes) / (sum.num_bytes + avail);
```

Comment:
@dzafman thanks for testing and the review.

but i think it would be enough to check the %USED with the expected value.

anyway, i tested by

1. creating a vstart cluster with 4 OSDs, and 
1. adding following crush rules to the default crush rule.
1. creating two pools `pool1` and `pool2` and setting crush_ruleset 1 and 2 for them respectively.
1. and setting the `sum.num_object_copies = 100` and set `sum.num_bytes = 1024*1024*1024*100` of pool1 using gdb
```
host host1 {
     id -3
     alg straw
     hash 0
     item osd.0 weight 1.000
     item osd.1 weight 1.000
}

host host2 {
     id -4
     alg straw
     hash 0
     item osd.2 weight 1.000
     item osd.3 weight 1.000
}

rule rep_rule1 {
    ruleset 1
    type replicated
    min_size 1
    max_size 10
    step take host1
    step choose firstn 0 type osd
    step emit
}

rule rep_rule2 {
    ruleset 2
    type replicated
    min_size 1
    max_size 10
    step take host2
    step choose firstn 0 type osd
    step emit
}
```

```
$ ./bin/ceph osd df
ID WEIGHT  REWEIGHT SIZE  USE  AVAIL %USE  VAR  PGS
 2 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  18
 3 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  18
 0 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  22
 1 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  14
 0 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  22
 1 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  14
 2 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  18
 3 1.00000  1.00000 1676G 198G 1477G 11.86 1.00  18
              TOTAL 6706G 795G 5910G 11.86
MIN/MAX VAR: 1.00/1.00  STDDEV: 0

$ ./bin/ceph df
GLOBAL:
    SIZE      AVAIL     RAW USED     %RAW USED
    6706G     5910G         795G         11.86
POOLS:
    NAME      ID     USED        %USED     MAX AVAIL     OBJECTS
    rbd       0            0         0         1970G           0
    pool1     1      102400M      9.22          985G           0
    pool2     2            0         0          985G           0
```

and the `%USED` of pool1 is expected:
```
In [4]: 102400/ (985*1024+102400)
Out[4]: 0.09216589861751152
```